packages/components/src/internal/components/forms/input/TextChoiceInput.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { ReactNode } from 'react';
+import { SelectInput, SelectInputProps } from './SelectInput';
+import { QueryColumn } from '../../../../public/QueryColumn';
+import { DisableableInput, DisableableInputState } from './DisableableInput';
+
+interface Props extends SelectInputProps {
+    queryColumn: QueryColumn;
+}
+
+export class TextChoiceInput extends DisableableInput<Props, DisableableInputState> {
+    render(): ReactNode {
+        const { queryColumn, placeholder, ...inputProps } = this.props;
+        const inputOptions = queryColumn.validValues.map(val => {
```

Comment:
Any reason to be defensive here about `validValues` not being defined?  I know there are checks around the usages of this, but if we somehow get here without having done that check do we want to return null? Or a `SelectInput` with an empty set of options?

Reply:
Good call. Done.